scripts/lint.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+set -eou pipefail
+
+# relative path calculation shim, I hope python is installed!
+function relative_path {
```

Comment:
we could just swap this out for sed instead?

Reply:
I believe osx sed and linux are slightly different, but perhaps not enough to matter in this case? I just was trying to minimize my cross OS stuff, but I can add a sed fallback if no python perhaps, I'll have to handle all the path separator logic myself potentially though, although I guess sed is probably not on windows so maybe a non-issue.

Reply:
yeah I was mostly thinking about Windows here if I'm honest, given most macOS and Linux configurations ship with a Python setup